client/client.go

```diff
@@ -218,14 +234,41 @@ func (c Client) RoleTargetsPath(roleName string, snapshotMeta data.Files, consis
 		}
 		roleName = path.Join(
 			dir,
-			fmt.Sprintf("%s.%s.json", roleMeta.Hashes["sha256"], roleName),
+			fmt.Sprintf("%s.%s.json", hasheSha256, roleName),
 		)
 	}
 	return roleName, nil
 }
 
-func (c Client) TargetMeta(path string) *data.FileMeta {
-	return c.local.FindTarget(path)
+// TargetMeta ensures the repo is up to date, downloading the minimum
+// necessary metadata files
+func (c Client) TargetMeta(path string) (*data.FileMeta, error) {
+	c.Update()
+	var meta *data.FileMeta
+
+	// FIFO list of targets delegations to inspect for target
+	roles := []string{data.ValidRoles["targets"]}
+	for ; len(roles) > 0; role := roles[0] {
```

Comment:
he's appending in the loop; i think that'll break the iterator if he just uses `range(roles)`